lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3Context.java

```diff
@@ -84,17 +86,27 @@ public void applyCredentialProviderOverride(AwsRequestOverrideConfiguration.Buil
         credentialsProviderOverride.ifPresent(builder::credentialsProvider);
     }
 
-    record S3SseContext(S3FileSystemConfig.S3SseType sseType, String sseKmsKeyId)
+    record S3SseContext(S3FileSystemConfig.S3SseType sseType, String sseKmsKeyId, S3SseCustomerKey sseCustomerKey)
     {
         public S3SseContext
         {
             requireNonNull(sseType, "sseType is null");
-            checkArgument((sseType != S3FileSystemConfig.S3SseType.KMS) || (sseKmsKeyId != null), "sseKmsKeyId is null for SSE-KMS");
+            if (sseType == KMS) {
+                checkArgument(sseKmsKeyId != null, "sseKmsKeyId is null for SSE-KMS");
+            }
+            if (sseType == CUSTOMER) {
+                checkArgument(sseCustomerKey != null, "sseCustomerKey is null for SSE-C");
+            }
+        }
+
+        public static S3SseContext of(S3FileSystemConfig.S3SseType sseType, String sseKmsKeyId, String sseCustomerKey)
```

Comment:
Why do we need an additional method `of` can't we use the constructor?

Reply:
I added to it to include  building object S3SseCustomerKey from String into the class - but as a preparation step.